node/network/statement-distribution/src/lib.rs

```diff
@@ -1330,15 +1347,33 @@ async fn handle_incoming_message<'a>(
 	// perform only basic checks before verifying the signature
 	// as it's more computationally heavy
 	if let Err(rep) = peer_data.check_can_receive(&relay_parent, &fingerprint, max_message_count) {
+		// This situation can happen when a peer's Seconded message was lost
+		// but we have received the Valid statement.
+		// So we check it once and then ignore repeated violation to avoid
+		// reputation change flood.
+		let mut need_report_peer = true;
+
+		if rep == COST_UNEXPECTED_STATEMENT_UNKNOWN_CANDIDATE {
```

Comment:
nit: Not keen on using this as a discriminator, this should be the exception.

Reply:
Yes, I was not sure it is a good idea, but `can_report` might return different outcomes, and some of them must be treated seriously (like a corrupted statement or apparent flood).

Reply:
The main issue is that it actually compares two strings in this case :(